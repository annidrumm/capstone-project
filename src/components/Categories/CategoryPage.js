import React from 'react'
import styled from 'styled-components/macro'
import forkIcon from '../../icons/fork.svg'
import gymIcon from '../../icons/gym.svg'
import beautyIcon from '../../icons/beauty.svg'
import fashionIcon from '../../icons/mannequin.svg'
import InteriorIcon from '../../icons/vase.svg'
import travelIcon from '../../icons/plane.svg'

export default function CategoryPage() {
  return (
    <ContainerStyled>
      <HeadlineStyled>
        What category are<br></br> you looking for?
      </HeadlineStyled>
      <section>
        <ButtonStyled category="food">
          <IconsStyled src={forkIcon} alt="" />
          <ButtonTextStyled>Food</ButtonTextStyled>
        </ButtonStyled>
        <ButtonStyled category="fitness">
          <IconsStyled src={gymIcon} alt="" />
          <ButtonTextStyled>Fitness</ButtonTextStyled>
        </ButtonStyled>
      </section>
      <section>
        <ButtonStyled category="beauty">
          <img src={beautyIcon} alt="" />
          <ButtonTextStyled>Beauty</ButtonTextStyled>
        </ButtonStyled>
        <ButtonStyled category="fashion">
          <img src={fashionIcon} alt="" />
          <ButtonTextStyled>Fashion</ButtonTextStyled>
        </ButtonStyled>
      </section>
      <section>
        <ButtonStyled category="interior">
          <img src={InteriorIcon} alt="" />
          <ButtonTextStyled>Interior</ButtonTextStyled>
        </ButtonStyled>
        <ButtonStyled category="travel">
          <img src={travelIcon} alt="" />
          <ButtonTextStyled>Travel</ButtonTextStyled>
        </ButtonStyled>
      </section>
    </ContainerStyled>
  )
}

const ContainerStyled = styled.section`
  text-align: center;
`

const HeadlineStyled = styled.h1`
  font: var(--font-weight-regular);
  font-size: 35px;
`
const ButtonStyled = styled.button`
  width: 128px;
  height: 101px;
  background: transparent;
  margin: 10px;
  border-radius: 10%;

  border-color: ${({ category }) =>
    category === 'food'
      ? 'var(--darkyellow)'
      : category === 'fitness'
      ? 'var(--lightpink)'
      : category === 'beauty'
      ? 'var(--lightgreen)'
      : category === 'fashion'
      ? 'var(--lightblue)'
      : category === 'interior'
      ? 'var(--lightpurple)'
      : 'var(--darkyellow)'};
`
const IconsStyled = styled.img`
  margin-top: 15px;
  vertical-align: top;
`
const ButtonTextStyled = styled.h2`
  color: var(--font-color-white);
  font-size: 20px;
  font-weight: var(--font-weight-light);
  vertical-align: bottom;
  margin-bottom: 7px;
  margin-top: 3px;
`
