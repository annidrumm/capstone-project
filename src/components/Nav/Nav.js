import React from 'react'
import styled from 'styled-components/macro'
import GoBackArrow from '../../components/GoBackArrow/GoBackArrow'
import { ReactComponent as HeartIcon } from '../../icons/heart.svg'
import { Switch, Route, Link } from 'react-router-dom'

export default function NavHeader() {
  return (
    <>
      <GoBackArrow />
      <Switch>
        <Route>
          <Link to="/favourites">
            <ButtonStyled>
              <HeartIcon fill="var(--darkpink)" stroke="var(--darkpink)" />
            </ButtonStyled>
          </Link>
        </Route>
      </Switch>
    </>
  )
}

const ButtonStyled = styled.button`
  background: transparent;
  border-radius: 50%;
  text-align: center;
  height: 60px;
  width: 60px;
  display: block;
  margin: auto;
  margin-top: 15px;
  border: none;
  cursor: pointer;
`
