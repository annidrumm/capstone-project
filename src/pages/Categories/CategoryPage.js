import React from 'react'
import styled from 'styled-components/macro'
import Headline from '../../components/Headline/Headline'
import CategoryButton from '../../components/CategoryButton/CategoryButton'

export default function CategoryPage() {
  return (
    <ContainerStyled>
      <Headline headline="What categories are you looking for?" />
      <CategoryButton />
    </ContainerStyled>
  )
}

const ContainerStyled = styled.section`
  text-align: center;
`
