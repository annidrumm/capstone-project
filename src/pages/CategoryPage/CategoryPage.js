import React from 'react'
import styled from 'styled-components/macro'
import Headline from '../../components/Headline/Headline'
import CategoryButtons from '../../components/CategoryButtons/CategoryButtons'
import profile1 from '../../images/nico-marks-min.jpg'
import profile2 from '../../images/fezbot2000-min.jpg'
import profile3 from '../../images/any-haust-min.jpg'
import profile4 from '../../images/ian-dooleyw-min.jpg'
import profile5 from '../../images/svyatoslav-romanov-min.jpg'

export default function CategoryPage() {
  return (
    <>
      <ContainerStyled>
        <Headline headline="What categories are you looking for?" />
        <CategoryButtons />
      </ContainerStyled>
      <SlideStyled>
        <ImageStyled src={profile1} />
        <ImageStyled src={profile2} />
        <ImageStyled src={profile3} />
        <ImageStyled src={profile4} />
        <ImageStyled src={profile5} />
      </SlideStyled>
    </>
  )
}

const ContainerStyled = styled.section`
  text-align: center;
`

const SlideStyled = styled.section`
  background: var(--transparent-blue);
  height: 110px;
  position: fixed;
  left: 0;
  bottom: 0;
  width: 100%;
  overflow-x: scroll;
  overflow-y: hidden;
  white-space: nowrap;
  @media (min-width: 1170px) {
    height: 100px;
  }
`
const ImageStyled = styled.img`
  width: 80px;
  height: 100px;
  padding: 15px;
  border-radius: 50%;
  margin: 0 10px 10px 10px;

  @media (min-width: 400px) {
    width: 100px;
    height: 120px;
    border-radius: 50%;
    margin: 0 10px 10px 30px;
  }
  @media (min-width: 650px) {
    width: 100px;
    height: 120px;
    border-radius: 50%;
    margin: 0 10px 20px 50px;
  }
  @media (min-width: 800px) {
    width: 100px;
    height: 120px;
    border-radius: 50%;
    margin: 0 10px 20px 100px;
  }
  @media (min-width: 800px) {
    width: 100px;
    height: 120px;
    border-radius: 50%;
    margin: 0 10px 20px 100px;
  }
  @media (min-width: 1170px) {
    width: 110px;
    height: 130px;
    border-radius: 50%;
    margin: 0 100px 100px 100px;
  }
`
