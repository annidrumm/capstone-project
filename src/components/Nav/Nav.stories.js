import React from 'react'
import Nav from './Nav.js'

export default {
  title: 'Components/Navigation',
  component: Nav,
}

export const navigation = () => {
  return <Nav />
}
