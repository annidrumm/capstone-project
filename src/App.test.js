import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import { MemoryRouter } from 'react-router-dom'
import CategoryPage from './pages/CategoryPage/CategoryPage'
import Nav from './components/Nav/Nav'

describe('App', () => {
  it('has links to all pages and displays them', () => {
    const urls = [
      '/',
      '/InfluencerList',
      '/food',
      '/fitness',
      '/beauty',
      '/fashion',
      '/interior',
      '/travel',
      '/favourites',
    ]
    const { getAllByRole } = render(
      <MemoryRouter>
        <CategoryPage />
        <Nav />
      </MemoryRouter>
    )
    getAllByRole('link').forEach((link) => {
      expect(urls).toContain(link.getAttribute('href'))
    })
  })
})

it('renders App correctly', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <CategoryPage />
      <Nav />
    </MemoryRouter>
  )
  expect(tree).toMatchSnapshot()
})
