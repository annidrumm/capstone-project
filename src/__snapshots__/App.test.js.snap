// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories renders correctly 1`] = `
<section
  className="CategoryPage__ContainerStyled-inp28z-0 gAaNNM"
>
  <section
    className="sc-AxjAm cmCotq"
  >
    <img
      alt="arrow left"
      className="sc-AxiKw eHrRzm"
      onClick={[Function]}
      src="back.svg"
    />
    <h1
      className="sc-AxirZ hykbjA"
    >
      What categories are you looking for?
    </h1>
  </section>
  <section
    className="CategoryButtons__ContainerStyled-fe33yf-0 kSPRRI"
  >
    <section>
      <button
        className="CategoryButtons__ButtonStyled-fe33yf-1 jdljUJ"
        onClick={[Function]}
      >
        <img
          alt="fork and spoon"
          className="CategoryButtons__IconsStyled-fe33yf-2 gUDOhC"
          src="fork.svg"
        />
        <h2
          className="CategoryButtons__H2Styled-fe33yf-3 gDPoMv"
        >
          Food
        </h2>
      </button>
      <button
        className="CategoryButtons__ButtonStyled-fe33yf-1 fSsSAP"
        onClick={[Function]}
      >
        <img
          alt="dumbbell"
          className="CategoryButtons__IconsStyled-fe33yf-2 gUDOhC"
          src="gym.svg"
        />
        <h2
          className="CategoryButtons__H2Styled-fe33yf-3 gDPoMv"
        >
          Fitness
        </h2>
      </button>
    </section>
    <section>
      <button
        className="CategoryButtons__ButtonStyled-fe33yf-1 eFVMfs"
        onClick={[Function]}
      >
        <img
          alt="lipsticks"
          src="beauty.svg"
        />
        <h2
          className="CategoryButtons__H2Styled-fe33yf-3 gDPoMv"
        >
          Beauty
        </h2>
      </button>
      <button
        className="CategoryButtons__ButtonStyled-fe33yf-1 bklmwB"
        onClick={[Function]}
      >
        <img
          alt="fashion mannequin"
          src="mannequin.svg"
        />
        <h2
          className="CategoryButtons__H2Styled-fe33yf-3 gDPoMv"
        >
          Fashion
        </h2>
      </button>
    </section>
    <section>
      <button
        className="CategoryButtons__ButtonStyled-fe33yf-1 bLieCZ"
        onClick={[Function]}
      >
        <img
          alt="vase"
          src="vase.svg"
        />
        <h2
          className="CategoryButtons__H2Styled-fe33yf-3 gDPoMv"
        >
          Interior
        </h2>
      </button>
      <button
        className="CategoryButtons__ButtonStyled-fe33yf-1 jdljUJ"
        onClick={[Function]}
      >
        <img
          alt="plane"
          src="plane.svg"
        />
        <h2
          className="CategoryButtons__H2Styled-fe33yf-3 gDPoMv"
        >
          Travel
        </h2>
      </button>
    </section>
  </section>
</section>
`;
